app/scripts/lib/marketing-email-client.js

```diff
@@ -7,38 +7,45 @@
  */
 
 define([
+  'lib/constants',
   'lib/xhr',
   'lib/marketing-email-errors'
-], function (xhr, MarketingEmailErrors) {
+], function (Constants, xhr, MarketingEmailErrors) {
   'use strict';
 
   function MarketingEmailClient(options) {
     options = options || {};
 
     var self = this;
-
+    self._xhrTimeout = options.timeout || Constants.DEFAULT_XHR_TIMEOUT_MS;
     self._xhr = options.xhr || xhr;
     self._baseUrl = options.baseUrl;
     self._preferencesUrl = options.preferencesUrl;
   }
 
   MarketingEmailClient.prototype = {
+    _normalizeBasketXHRError: function (xhr) {
+      var respJSON = xhr.responseJSON;
+      if (respJSON && ! ('errno' in respJSON) && respJSON.code) {
+        // the basket API returns code instead of errno. Convert.
+        respJSON.errno = respJSON.code;
+        // we want to set the code equal to the http status code
+        respJSON.code = xhr.status;
+      }
+    },
+
     _request: function (method, endpoint, accessToken, data) {
       var url = this._baseUrl + endpoint;
-
+      var self = this;
       return this._xhr.oauthAjax({
         url: url,
         type: method,
         accessToken: accessToken,
-        data: data
+        data: data,
+        timeout: this._xhrTimeout
       })
       .fail(function (xhr) {
-        var respJSON = xhr.responseJSON;
-        if (respJSON && ! ('errno' in respJSON) && respJSON.code) {
-          // the basket API returns code instead of errno. Convert.
-          respJSON.errno = respJSON.code;
-        }
-
+        self._normalizeBasketXHRError(xhr);
```

Comment:
We are being kinda silly here. `MarketingEmailErrors` is based on errors.js, which already has a `normalizeXHRError`. `MarketingEmailErrors` should override `normalizeXHRError`. The overridden function will perform Basket specific normalization, then call the super's `normalizeXHRError`